src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowLoopSpec.groovy

```diff
@@ -0,0 +1,618 @@
+        withPool {
+            [exam.forward, exam.reverse].eachParallel { direction ->
+                def resources = traffExam.startExam(direction)
+                direction.setResources(resources)
+                assert traffExam.waitExam(direction).hasTraffic()
+            }
+        }
+
+        when: "Create flowLoop on the dst switch"
+        northboundV2.createFlowLoop(flow.flowId, new FlowLoopPayload(switchPair.dst.dpId))
+
+        then: "FlowLoop is created on the dst switch"
+        //there is a bug, no flow_loop info in getFlowV2
+        with(northbound.getFlow(flow.flowId)) {
+            it.looped
+            it.loopSwitchId == switchPair.dst.dpId
+        }
+
+        and: "FlowLoop rules are created"
+        Wrappers.wait(RULES_INSTALLATION_TIME) {
+            assert northbound.getSwitchRules(switchPair.dst.dpId).flowEntries.findAll {
+                new Cookie(it.cookie).getType() == CookieType.FLOW_LOOP
+            }.size() == 2
+        }
+
+        when: "Delete the flow"
+        flowHelperV2.deleteFlow(flow.flowId)
+
+        then: "FlowLoop rules are deleted on the dst switch"
+        assert northbound.getSwitchRules(switchPair.dst.dpId).flowEntries.findAll {
+            new Cookie(it.cookie).getType() == CookieType.FLOW_LOOP
+        }.empty
+
+        and: "The dst switch is valid"
+        northbound.validateSwitch(switchPair.src.dpId).verifyRuleSectionsAreEmpty(["missing", "excess", "misconfigured"])
+        def testIsCompleted = true
+
+        cleanup: "Delete the flow"
+        if (!testIsCompleted) {
+            flowHelperV2.deleteFlow(flow.flowId)
+            [switchPair.src.dpId, switchPair.dst.dpId].each {
+                northbound.synchronizeSwitch(it, true)
+            }
+        }
+
+        where:
+        data << [[
+                         flowDescription: "pinned",
+                         switchPair     : { List<SwitchId> switchIds ->
+                             getSwPairConnectedToTraffGenForSimpleFlow(switchIds)
+                         },
+                         flow           : { FlowRequestV2 fl -> fl.tap { pinned = true } }
+                 ],
+                 [
+                         flowDescription: "default",
+                         switchPair     : { List<SwitchId> switchIds ->
+                             getSwPairConnectedToTraffGenForSimpleFlow(switchIds)
+                         },
+                         flow           : { FlowRequestV2 fl ->
+                             fl.tap {
+                                 source.vlanId = 0
+                                 destination.vlanId = 0
+                             }
+                         }
+                 ],
+                 [
+                         flowDescription: "protected",
+                         switchPair     : { List<SwitchId> switchIds ->
+                             getSwPairConnectedToTraffGenForProtectedFlow(switchIds)
```

Comment:
I think interaction with protected flows is not fully revealed here? What happens when main path breaks and flow path falls to protected?